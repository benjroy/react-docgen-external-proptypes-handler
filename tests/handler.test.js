const fs = require('fs');
const path = require('path');
const glob = require('glob');
const docgen = require('react-docgen');
const original_externalProptypesHandler = require("../index");
// const externalPropTypeHandler = require("../externalPropTypeHandler").propTypeHandler;
const nightly_externalPropTypeHandler = require("../nightly_externalPropTypeHandler").propTypeHandler;

const { expect } = global;

const CWD = process.cwd();

describe('react-docgen-external-proptypes-handler', () => {
  // const HANDLER = original_externalProptypesHandler;
  // const HANDLER = externalPropTypeHandler;
  const HANDLER = nightly_externalPropTypeHandler;

  function processFilepathWithHandlers(filepath, handlers) {
    const resolver = docgen.resolver.findExportedComponentDefinition;
    const options = {};
    // read file to get source code
    const source = fs.readFileSync(filepath, 'utf8');
    return docgen.parse(source, resolver, handlers, options);
  }

  function processFilepathWithDefaultHandlers(filepath) {
    const handlers = docgen.defaultHandlers.concat(
      HANDLER(filepath)
    );
    return processFilepathWithHandlers(filepath, handlers);
  };

  function processFilepathWithHandler(filepath) {
    const handlers = ([]).concat(
      HANDLER(filepath)
    );
    return processFilepathWithHandlers(filepath, handlers);
  };

  function test(filepath) {
    // it('parses with default resolver/handlers and external handler for ${filepath}', () => {
    //   const info = processFilepathWithDefaultHandlers(filepath);
    //   expect(info).toMatchSnapshot();
    // });
    
    it('parses with only external handlers', () => {
      const info = processFilepathWithHandler(filepath);
      expect(info).toMatchSnapshot();
    });
  }


  const TARGET_FILES = glob.sync(path.resolve(CWD, 'tests/fixtures/components/component-*.js'));
  // const TARGET_FILES = glob.sync(path.resolve(CWD, 'tests/fixtures/components/component-spreads-call-expression.js'));
  // const TARGET_FILES = glob.sync(path.resolve(CWD, 'tests/fixtures/components/component-spreads-external-member-expression.js'));
  // const TARGET_FILES = glob.sync(path.resolve(CWD, 'tests/fixtures/components/component-imports-api-cjs*.js'));
  // const TARGET_FILES = glob.sync(path.resolve(CWD, 'tests/fixtures/components/component-spread*.js'));
  // const TARGET_FILES = glob.sync(path.resolve(CWD, 'tests/fixtures/components/component-no-*.js'));
  // const TARGET_FILES = glob.sync(path.resolve(CWD, 'tests/fixtures/components/component-local-*.js'));
  // const TARGET_FILES = glob.sync(path.resolve(CWD, 'tests/fixtures/components/component-spreads-nested.js'));
  // const TARGET_FILES = glob.sync(path.resolve(CWD, 'tests/fixtures/components/component-{spreads-nested,literal}.js'));
  // const TARGET_FILES = glob.sync(path.resolve(CWD, 'tests/fixtures/components/component-literal.js'));
  // const TARGET_FILES = glob.sync(path.resolve(CWD, 'tests/fixtures/components/component-imports-external-*.js'));
  // const TARGET_FILES = glob.sync(path.resolve(CWD, 'tests/fixtures/components/component-imports-external-prop*.js'));
  // const TARGET_FILES = glob.sync(path.resolve(CWD, 'tests/fixtures/components/component-imports-literal*.js'));
  // const TARGET_FILES = glob.sync(path.resolve(CWD, 'tests/fixtures/components/component-imports-literal-default.js'));
  // const TARGET_FILES = glob.sync(path.resolve(CWD, 'tests/fixtures/components/component-*-enum*.js'));

  TARGET_FILES.forEach(filepath => {
    describe(filepath, () => test(filepath));
  });


});

