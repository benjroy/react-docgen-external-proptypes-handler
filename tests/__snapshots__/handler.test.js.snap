// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-docgen-external-proptypes-handler processes all props for component-imports-external-values-for-enum 1`] = `
Object {
  "description": "",
  "displayName": "ComponentImportsExternalValuesForEnum",
  "methods": Array [],
}
`;

exports[`react-docgen-external-proptypes-handler processes all props for component-imports-literal 1`] = `
Object {
  "composes": Array [
    "./api-1",
  ],
  "description": "",
  "displayName": "ComponentImportsLiteral",
  "methods": Array [],
  "props": Object {
    "api-1-bool": Object {
      "description": "Comment for api-1-bool.",
      "required": false,
      "type": Object {
        "name": "bool",
      },
    },
    "api-1-node-required": Object {
      "description": "Comment for api-1-node-required.",
      "required": true,
      "type": Object {
        "name": "node",
      },
    },
  },
}
`;

exports[`react-docgen-external-proptypes-handler processes all props for component-local-spread 1`] = `
Object {
  "description": "",
  "displayName": "ComponentLocalSpread",
  "methods": Array [],
  "props": Object {
    "bool": Object {
      "required": false,
      "type": Object {
        "name": "bool",
      },
    },
    "enum": Object {
      "required": false,
      "type": Object {
        "name": "enum",
        "value": Array [
          Object {
            "computed": false,
            "value": "1",
          },
          Object {
            "computed": false,
            "value": "2",
          },
          Object {
            "computed": false,
            "value": "3",
          },
        ],
      },
    },
    "enum-var-of-vars": Object {
      "required": false,
      "type": Object {
        "name": "enum",
        "value": Array [
          Object {
            "computed": false,
            "value": "1",
          },
          Object {
            "computed": false,
            "value": "'two'",
          },
          Object {
            "computed": false,
            "value": "3",
          },
        ],
      },
    },
    "enum-vars": Object {
      "required": false,
      "type": Object {
        "name": "enum",
        "value": Array [
          Object {
            "computed": false,
            "value": "1",
          },
          Object {
            "computed": false,
            "value": "'two'",
          },
          Object {
            "computed": false,
            "value": "3",
          },
        ],
      },
    },
    "node": Object {
      "required": true,
      "type": Object {
        "name": "node",
      },
    },
    "number": Object {
      "required": false,
      "type": Object {
        "name": "number",
      },
    },
    "shape": Object {
      "required": false,
      "type": Object {
        "name": "shape",
        "value": Object {
          "LOCAL_ENUM": Object {
            "name": "enum",
            "required": false,
            "value": Array [
              Object {
                "computed": false,
                "value": "1",
              },
              Object {
                "computed": false,
                "value": "'two'",
              },
              Object {
                "computed": false,
                "value": "3",
              },
            ],
          },
          "LOCAL_ONE": Object {
            "name": "number",
            "required": false,
          },
        },
      },
    },
    "string": Object {
      "required": false,
      "type": Object {
        "name": "string",
      },
    },
  },
}
`;

exports[`react-docgen-external-proptypes-handler processes all props for component-no-external-spread 1`] = `
Object {
  "description": "",
  "displayName": "ComponentNoExternalSpread",
  "methods": Array [],
  "props": Object {
    "local-bool-spread": Object {
      "required": true,
      "type": Object {
        "name": "bool",
      },
    },
    "local-enum": Object {
      "defaultValue": Object {
        "computed": false,
        "value": "3",
      },
      "description": "Comment for local-enum.",
      "required": false,
      "type": Object {
        "name": "enum",
        "value": Array [
          Object {
            "computed": false,
            "value": "9",
          },
          Object {
            "computed": false,
            "value": "'eight'",
          },
          Object {
            "computed": false,
            "value": "7",
          },
        ],
      },
    },
    "local-enum-external-list": Object {
      "description": "",
      "required": false,
      "type": Object {
        "computed": true,
        "name": "enum",
        "value": "ENUM",
      },
    },
    "local-enum-external-values": Object {
      "description": "",
      "required": false,
      "type": Object {
        "name": "enum",
        "value": Array [
          Object {
            "computed": true,
            "value": "VALUES.ZERO",
          },
          Object {
            "computed": true,
            "value": "VALUES.ONE",
          },
          Object {
            "computed": true,
            "value": "VALUES.TWO",
          },
          Object {
            "computed": true,
            "value": "VALUES.THREE",
          },
        ],
      },
    },
    "local-node": Object {
      "description": "Comment for local-node.",
      "required": true,
      "type": Object {
        "name": "node",
      },
    },
    "local-number": Object {
      "description": "Comment for local-number.",
      "required": false,
      "type": Object {
        "name": "number",
      },
    },
    "local-string": Object {
      "defaultValue": Object {
        "computed": false,
        "value": "'default-local-string'",
      },
      "description": "Comment for local-string.",
      "required": false,
      "type": Object {
        "name": "string",
      },
    },
  },
}
`;

exports[`react-docgen-external-proptypes-handler processes all props for component-spreads 1`] = `
Object {
  "composes": Array [
    "./api-1",
    "./api-2",
  ],
  "description": "",
  "displayName": "ComponentSpreads",
  "methods": Array [],
  "props": Object {
    "api-1-bool": Object {
      "description": "Comment for api-1-bool.",
      "required": false,
      "type": Object {
        "name": "bool",
      },
    },
    "api-1-node-required": Object {
      "description": "Comment for api-1-node-required.",
      "required": true,
      "type": Object {
        "name": "node",
      },
    },
    "api-2-node-required": Object {
      "description": "Comment for api-2-node-required.",
      "required": true,
      "type": Object {
        "name": "node",
      },
    },
    "api-2-string": Object {
      "description": "Comment for api-2-string.",
      "required": false,
      "type": Object {
        "name": "string",
      },
    },
    "local-string": Object {
      "defaultValue": Object {
        "computed": false,
        "value": "'default-local-string'",
      },
      "description": "",
      "required": false,
      "type": Object {
        "name": "string",
      },
    },
  },
}
`;

exports[`react-docgen-external-proptypes-handler processes all props for component-spreads-nested 1`] = `
Object {
  "composes": Array [
    "./api-3",
  ],
  "description": "",
  "displayName": "ComponentSpreadsNested",
  "methods": Array [],
  "props": Object {
    "api-1-bool": Object {
      "description": "Comment for api-1-bool.",
      "required": false,
      "type": Object {
        "name": "bool",
      },
    },
    "api-1-node-required": Object {
      "description": "Comment for api-1-node-required.",
      "required": true,
      "type": Object {
        "name": "node",
      },
    },
    "api-2-node-required": Object {
      "description": "Comment for api-2-node-required.",
      "required": true,
      "type": Object {
        "name": "node",
      },
    },
    "api-2-string": Object {
      "description": "Comment for api-2-string.",
      "required": false,
      "type": Object {
        "name": "string",
      },
    },
    "api-3-node-required": Object {
      "description": "Comment for api-3-node-required.",
      "required": true,
      "type": Object {
        "name": "node",
      },
    },
    "api-3-number": Object {
      "description": "Comment for api-3-number.",
      "required": false,
      "type": Object {
        "name": "number",
      },
    },
    "local-number": Object {
      "defaultValue": Object {
        "computed": false,
        "value": "4242",
      },
      "description": "",
      "required": false,
      "type": Object {
        "name": "number",
      },
    },
  },
}
`;
